### ISSUE TITLE
MD076: treat block quotes similar to fences code blocks in tight lists

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#523]
### Problem or motivation

When looking at Markdown with fenced code blocks and block quotes in lists, they look quite similar.

However, MD076 (https://github.com/rvben/rumdl/blob/a00435eea5c100673c59a99e3e434825992a90c1/docs/md076.md) allows for whitespace around fenced code blocks in tight lists but not around block quotes. This Markdown sample illustrates:

````markdown
---
title: Heading
---

Some introductory text:

- This is a list item with a code block:

  ```markdown
  Some code
  ```

- This is a list item with a block quote:

  > [!note]
  > This is a note

- This is another list item.
````

This sample will produce the error:

```shell
test.md:17:1: [MD076] Unexpected blank line between list items
```

Though fenced code blocks aren't the same as block quotes (and don't have a rule like MD031 - https://github.com/rvben/rumdl/blob/a00435eea5c100673c59a99e3e434825992a90c1/docs/md031.md) for them, it might still be helpful if block quotes could be treated similarly to fenced code blocks in lists.

### Proposed solution

MD076 be modified so that the Markdown sample above passes when `style = 'tight'`.

### Alternatives considered

I see a couple of other possibilities:

- MD076 could be made configurable so that block quotes are allowed to have spaces around them in lists
- A new rule could be made for block quotes similar to MD031. Something like _Block quotes need blank lines around them_ which then something like MD076 could enforce in lists.

### Feature category

Improve existing rule

### Example markdown

```markdown
---
title: Heading
---

Some introductory text:

- This is a list item with a code block:

  
  Some code
  

- This is a list item with a block quote:

  > [!note]
  > This is a note

- This is another list item.
```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature

### Additional context

Required configuration:

```toml
[MD076]
style = 'tight'
```

**Comment by eread** [comment]
CC @Ravlen @axilleas just FYI.

**Comment by eread** [comment]
@rvben The more I think about it, the more I think the "new rule alternative" would be good!

If it was modelled on MD031, then this Markdown style could be enforced:

````markdown
---
title: Heading
---

This is a fenced code block:

```markdown
This some text
```

This is a block quote:

> This some text

Some introductory text:

- This a list item with a fenced code block:

  ```markdown
  This some text
  ```

- This a list item with a block quote:

  > This some text

- This is the last list item.

This is more text.
````

And this style could be disallowed:

````markdown
---
title: Heading
---

This is a fenced code block:
```markdown
This some text
```
This is a block quote:
> This some text
Some introductory text:

- This a list item with a fenced code block:
  ```markdown
  This some text
  ```
- This a list item with a block quote:
  > This some text
- This is the last list item.

This is more text.
```

WDYT?

**Comment by rvben** [comment]
Fixed in v0.1.49. Blockquotes inside tight lists are now treated as structural content, preventing false positive spacing warnings. Thanks for reporting, @eread!
